### PR TITLE
Minor perf improvements..

### PR DESCRIPTION
**Pull request by cijothomas**
Fix Issue # .
<Short description of the fix.>

- [ ] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue.
- [ ] The PR will trigger build, unit tests, and functional tests automatically. If your PR was submitted from fork - mention one of committers to initiate the build for you.
	  If you want to to re-run the build/tests, the easiest way is to simply Close and Re-Open this same PR. (Just click 'close pull request' followed by 'open pull request' buttons at the bottom of the PR)

- Please follow [these] (https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### CODE REVIEW

File: src/Microsoft.ApplicationInsights/TelemetryClient.cs [hunk 1]
```diff
@@ -478,20 +478,20 @@ public void InitializeInstrumentationKey(ITelemetry telemetry)
...
-            }
-
             ISupportAdvancedSampling telemetryWithSampling = telemetry as ISupportAdvancedSampling;
 
             // Telemetry can be already sampled out if that decision was made before calling Track()
             bool sampledOut = telemetryWithSampling?.IsSampledOutAtHead ?? false;
 
             if (!sampledOut)
             {
+                string instrumentationKey = this.Context.InstrumentationKey;
```
Comment (MS-TimothyMothra) [inline src/Microsoft.ApplicationInsights/TelemetryClient.cs]: nit: Consider moving this down to line 523 where the instrumentationKey is being used.

Comment (cijothomas) [reply src/Microsoft.ApplicationInsights/TelemetryClient.cs]: agree

File: src/Microsoft.ApplicationInsights/TelemetryClient.cs [hunk 2]
```diff
@@ -520,6 +513,13 @@ public void Initialize(ITelemetry telemetry)
                     Utils.CopyDictionary(this.Context.GlobalProperties, telemetry.Context.GlobalProperties);
                 }
 
+                string instrumentationKey = this.Context.InstrumentationKey;
+
+                if (string.IsNullOrEmpty(instrumentationKey))
+                {
+                    instrumentationKey = this.configuration.InstrumentationKey;
```
Comment (Dmitry-Matveev) [inline src/Microsoft.ApplicationInsights/TelemetryClient.cs]: I remember keeping this assignment outside to make sure that proactively sampled out item won't get "Invalid Ikey" or won't get dropped before it arrives to sampling for gain up calculation in Sampling Telemetry Processor. If we are sure that items make it all the way without IKey - this change is OK. There was a case where setting empty IKey prevented all telemetry from leaving SDK (I may have false memory here, though).

Comment (cijothomas) [reply src/Microsoft.ApplicationInsights/TelemetryClient.cs]: The lines i moved into the if block will not have helped with the issue. It just assigns ikey to a local variable, which is ONLY used inside the if block. I just moved the assignment closer to actual usage, in the same if block.

Comment (cijothomas) [reply src/Microsoft.ApplicationInsights/TelemetryClient.cs]: https://github.com/microsoft/ApplicationInsights-aspnetcore/blob/develop/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs#L513 This initialization is probably what you are referring to. That assignment is outside of any Sampling If-Else statements, and is correctly done so,

**Pull request merged**